gensim/models/experimental/drmm_tks.py

```diff
@@ -0,0 +1,595 @@
+                    (n_non_embedding_words, self.vocab_size, n_non_embedding_words * 100 / self.vocab_size,
+                     self.unk_handle_method))
+
+        # The point where vocab words end
+        vocab_offset = self.vocab_size
+
+        if self.keep_full_embedding:
+            # Include embeddings for words in embedding file but not in the train vocab
+            # It will be useful for embedding words encountered in validation and test set
+            logger.info(
+                "Adding additional words from the embedding file to embedding matrix")
+            i = self.vocab_size
+            extra_embeddings = []
+            # Take the words in the embedding file which aren't there int the train vocab
+            for word in list(kv_model.vocab):
+                if word not in self.word2index:
+                    # Add the new word's vector and index it
+                    extra_embeddings.append(kv_model[word])
+                    # We also need to keep an additional indexing of these
+                    # words
+                    self.word2index[word] = i
+                    i += 1
+
+            vocab_offset = i
+
+        # Set the pad and unk word to second last and last index
+        self.pad_word_index = vocab_offset
+        self.unk_word_index = vocab_offset + 1
+
+        if self.unk_handle_method == 'random':
+            unk_embedding_row = np.random.uniform(
+                -0.2, 0.2, (1, self.embedding_dim))
+        elif self.unk_handle_method == 'zero':
+            unk_embedding_row = np.zeros((1, self.embedding_dim))
+
+        pad_embedding_row = np.random.uniform(-0.2,
+                                              0.2, (1, self.embedding_dim))
+
+        if self.keep_full_embedding:
+            self.embedding_matrix = np.vstack(
+                [self.embedding_matrix, np.array(extra_embeddings),
+                 pad_embedding_row, unk_embedding_row])
+        else:
+            self.embedding_matrix = np.vstack(
+                [self.embedding_matrix, pad_embedding_row, unk_embedding_row])
+
+        if self.normalize_embeddings:
+            logger.info("Normalizing the word embeddings")
+            self.embedding_matrix = normalize(self.embedding_matrix)
+
+        logger.info("Embedding Matrix build complete. It now has shape %s" %
+                    str(self.embedding_matrix.shape))
+        logger.info("Pad word has been set to index %d" % self.pad_word_index)
+        logger.info("Unknown word has been set to index %d" %
+                    self.unk_word_index)
+        logger.info("Embedding index build complete")
+
+    def make_indexed(self, sentence):
+        """Returns the indexed version of the sentence based on self.word2index
+        in the form of a list
+
+        Parameters:
+        -----------
+        sentence str
+            The sentence to be indexed
+
+        Raises:
+        -------
+        ValueError: If the sentence has a lenght more than text_maxlen
```

Comment:
the empty line at the end of docstring always (except one-line docstring case), here and everywhere.